firmware/thruna/BMS/Src/main.c

```diff
@@ -312,15 +313,40 @@ int main(void)
     airs = App_Airs_Create(
         Io_Airs_IsAirPositiveClosed, Io_Airs_IsAirNegativeClosed, Io_Airs_CloseAirPositive, Io_Airs_OpenAirPositive);
 
+    uint16_t soc_address;
+
+    eeprom = App_Eeprom_Create(Io_Eeprom_WritePage, Io_Eeprom_ReadPage, Io_Eeprom_PageErase);
+
+    // A negative soc value will indicate to App_SocStats_Create that saved SOC value is corrupted
+    float saved_soc_c = -1.0f;
+    if (App_Eeprom_ReadAddress(eeprom, &soc_address) == EXIT_CODE_OK)
+    {
+        if (App_Eeprom_ReadMinSoc(eeprom, soc_address, &saved_soc_c) == EXIT_CODE_OK)
+            ;
```

Comment:
why don't you just handle  
if (App_Eeprom_ReadMinSoc(eeprom, soc_address, &saved_soc_c) != EXIT_CODE_OK) case instead of having ; here

Reply:
agreed
also this logic is probably better handled in `App_Eeprom_Create` or some other EEPROM file initialization function

Reply:
Moved to App_SocStats_Create